table/tables/cache.go

```diff
@@ -30,74 +35,152 @@ var _ table.CachedTable = &cachedTable{}
 
 type cachedTable struct {
 	TableCommon
-	kv.MemBuffer
-	mu sync.RWMutex
+	cacheData atomic.Value
+	mu        sync.RWMutex
+	handle    StateRemote
+}
+
+func leaseFromTS(ts uint64) uint64 {
+	const defaultLeaseDuration time.Duration = 3 * time.Second
```

Comment:
// TODO: make this configurable in the following PRs

I'll add this action item here https://github.com/pingcap/tidb/issues/25293